servers/rendering/renderer_rd/effects/vrs.cpp

```diff
@@ -103,8 +109,16 @@ void VRS::copy_vrs(RID p_source_rd_texture, RID p_dest_framebuffer, bool p_multi
 }
 
 Size2i VRS::get_vrs_texture_size(const Size2i p_base_size) const {
-	int32_t texel_width = RD::get_singleton()->limit_get(RD::LIMIT_VRS_TEXEL_WIDTH);
-	int32_t texel_height = RD::get_singleton()->limit_get(RD::LIMIT_VRS_TEXEL_HEIGHT);
+	int32_t texel_width = 0;
+	int32_t texel_height = 0;
+	if (RD::get_singleton()->has_feature(RD::SUPPORTS_FRAGMENT_DENSITY_MAP)) {
```

Comment:
So this kind of bolts down to a design choice that I do know the right way forward for and ties into remarks I made in earlier comments.

Especially once Metal and DirectX come into play, we're starting to make a lot of GPU specific choices outside of our driver that will make code like in these spots more and more complex. If we can we should internalise these choices.

Also we're making the assumption here that if the density map extension is available, thats the choice we're going with. That may be true for the Quest (where this enables performance improvements), but there is no telling if this is universally true. It is very possible that on other hardware we should prioritise the KHR over the EXT if both are available.

I think we should have a way to indirect which one is prefered, where at initialisation either one or the other is enabled, and on calling `RD::get_singleton()->limit_get(RD::LIMIT_VRS_TEXEL_WIDTH);` we return the correct value for the extension we use. And it will be future proof for when we have Metal and/or DirectX support.

This also ensures that we don't accidentally mix and match logic, we get a consistent choice made by the rendering driver.

We can't get away with some querying and specific logic (like whether our density map has rates or weights and whether X and Y are split into channels) but also that requirement can be queried from the rendering driver which then returns the correct requirements based on the chosen extension.